core/color.js

```diff
@@ -0,0 +1,467 @@
+				compatibilityMode: false,
+				defaultHexColorCode: '#000000',
+				/**
+				 * @deprecated For compatibility
+				 */
+				legacyParsing: function( colorCode ) {
+					var stringToHex = this._.matchStringToNamedColor( colorCode );
+					var rgbToHex = this._.convertRgbStringToHex( colorCode );
+					var hexToHex = this._.normalizeHex( colorCode );
+
+					//due to compatibility
+					var defaultValue = colorCode;
+
+					var finalHex  = stringToHex || rgbToHex || hexToHex || defaultValue;
+
+					this._.hexColorCode = finalHex;
+				},
+				matchStringToNamedColor: function( colorName ) {
+					var colorToHexObject = CKEDITOR.tools.style.Color.namedColors;
+					var resultCode = colorToHexObject[ colorName.toLowerCase() ] || null;
+
+					return resultCode;
+				},
+				convertRgbStringToHex: function( styleText ) {
+					var rgbRegExp = /(?:rgb\(\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)\s*\))/gi;
+
+					if ( styleText.match( rgbRegExp ) ) {
+						return styleText.replace( rgbRegExp, function( match, red, green, blue ) {
+							var color = [ red, green, blue ];
+							// Add padding zeros if the hex value is less than 0x10.
+							for ( var i = 0; i < 3; i++ ) {
+								color[ i ] = ( '0' + parseInt( color[ i ], 10 ).toString( 16 ) ).slice( -2 );
+							}
+							return '#' + color.join( '' );
+						} );
+					} else {
+						return null;
+					}
+				},
+				/**
+				* Normalizes hexadecimal notation so that the color string is always 6 characters long and lowercase.
+				*
+				* @param {String} hexColorCode String containing hex colors.
+				* @returns {String} The style data with hex colors normalized.
+				*/
+				normalizeHex: function( hexColorCode ) {
+					//also style string with hex color
+					var styleHexRegExp = /#(([0-9a-f]{3}){1,2})($|;|\s+)/gi;
+
+					if ( hexColorCode.match( styleHexRegExp ) ) {
+						return hexColorCode.replace( styleHexRegExp, function( match, hexColor, hexColorPart, separator ) {
+							var normalizedHexColor = hexColor.toLowerCase();
+
+							if ( normalizedHexColor.length == 3 ) {
+								var parts = normalizedHexColor.split( '' );
+								normalizedHexColor = [ parts[ 0 ], parts[ 0 ], parts[ 1 ], parts[ 1 ], parts[ 2 ], parts[ 2 ] ].join( '' );
+							}
+
+							return '#' + normalizedHexColor + separator;
+						} );
+					} else {
+						return null;
+					}
+
+				},
+				parseInput: function( colorCode ) {
+					var hexStringFromNamedColor = this._.matchStringToNamedColor( colorCode );
+
+					var hexFromHexString = this._.matchStringToHex( colorCode );
+
+					var hexFromRgbOrHsl = this._.rgbOrHslToHex( colorCode );
+
+					this._.hexColorCode = hexStringFromNamedColor || hexFromHexString || hexFromRgbOrHsl || this._.defaultHexColorCode;
+				},
+				rgbOrHslToHex: function( colorCode ) {
+					var colorFormat = colorCode.trim().slice( 0,3 ).toLowerCase();
+
+					if ( colorFormat !== 'rgb' && colorFormat !== 'hsl' ) {
+						return null;
+					}
+
+					//take values
```

Comment:
It's good to keep comments as full sentences. Also they should bring additional value over the code and be easy to understand and complement commented code. Which means they should be very explicit to help understand the code and not to clutter up the code.

Reply:
This comment was removed.